src/github.com/matrix-org/dendrite/clientapi/routing/register.go

```diff
@@ -43,9 +46,13 @@ const (
 	minPasswordLength = 8   // http://matrix.org/docs/spec/client_server/r0.2.0.html#password-based
 	maxPasswordLength = 512 // https://github.com/matrix-org/synapse/blob/v0.20.0/synapse/rest/client/v2_alpha/register.py#L161
 	maxUsernameLength = 254 // http://matrix.org/speculator/spec/HEAD/intro.html#user-identifiers TODO account for domain
+	sessionIDLength   = 24
 )
 
-var validUsernameRegex = regexp.MustCompile(`^[0-9a-zA-Z_\-./]+$`)
+var (
+	sessions           = make(map[string][]authtypes.LoginType) // Sessions and completed flow stages
```

Comment:
We'll want to prune these sessions so that we don't just slowly but surely fill up with old sessions.

Happy for you to just add a TODO and can deal with it in another PR.

Reply:
Yeah, good catch. Obviously removing when registering is completed is fine and dandy, but will need to do something about a session length for non-completed registrations.

Reply:
Actually, we'll probably want to keep around completed registrations for a while in case the client retries the last register request.

Reply:
You mean even after `completeRegistration()` has been carried out?

In that case we can just have a session length be the only way to prune sessions.